physical/aerospike/aerospike.go

```diff
@@ -0,0 +1,155 @@
+package aerospike
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"strings"
+
+	aero "github.com/aerospike/aerospike-client-go"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/physical"
+)
+
+const (
+	keyBin   = "keyBin"
+	valueBin = "valueBin"
+)
+
+// AerospikeBackend is a physical backend that stores data in Aerospike.
+type AerospikeBackend struct {
+	client    *aero.Client
+	namespace string
+	set       string
+	logger    log.Logger
+}
+
+// Verify AerospikeBackend satisfies the correct interface.
+var _ physical.Backend = (*AerospikeBackend)(nil)
+
+// NewAerospikeBackend constructs an AerospikeBackend backend.
+func NewAerospikeBackend(conf map[string]string, logger log.Logger) (physical.Backend, error) {
+	namespace := conf["namespace"]
+	set := conf["set"]
+
+	policy := aero.NewClientPolicy()
+	policy.User = conf["username"]
+	policy.Password = conf["password"]
+
+	port, err := strconv.Atoi(conf["port"])
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := aero.NewClientWithPolicy(policy, conf["hostname"], port)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AerospikeBackend{
```

Comment:
Since (per the docs), `set`/`namespace`/`hostname` are all required, should we be validating `!= ""` before successfully returning?

Reply:
Looks like `set` is optional, and both `namespace` and `hostname` now have default values so we should be good.